bokehjs/src/lib/core/bokeh_events.ts

```diff
@@ -16,23 +16,33 @@ export abstract class BokehEvent {
 
   /* prototype */ event_name: string
 
-  origin: HasProps | null = null
-
   to_json(): EventJSON {
     const {event_name} = this
     return {event_name, event_values: this._to_json()}
   }
 
+  protected _to_json(): JSON {
+    return {}
+  }
```

Comment:
Given this is an abstract class, this should be an abstract method. Then `DocumentReady` should provide the trivial implementation. Otherwise (currently), every `_to_json` implementation must have a call to `super._to_json`, i.e. not depend on the internal implementation detail of `BokehEvent._to_json`.